mode/sparql/sparql.js

```diff
@@ -33,6 +33,14 @@ CodeMirror.defineMode("sparql", function(config) {
                              "true", "false", "with",
                              "data", "copy", "to", "move", "add", "create", "drop", "clear", "load", "into"]);
   var operatorChars = /[*+\-<>=&|\^\/!\?]/;
+  var PN_CHARS_BASE =
+    "[A-Za-z\\u{00C0}-\\u{00D6}\\u{00D8}-\\u{00F6}\\u{00F8}-\\u{02FF}" +
+    "\\u{0370}-\\u{037D}\\u{037F}-\\u{1FFF}\\u{200C}-\\u{200D}\\u{2070}-\\u{218F}" +
+    "\\u{2C00}-\\u{2FEF}\\u{3001}-\\u{D7FF}\\u{F900}-\\u{FDCF}\\u{FDF0}-\\u{FFFD}\\u{10000}-\\u{EFFFF}]";
+  var PN_CHARS_U = PN_CHARS_BASE + "|_";
+  var PN_CHARS = PN_CHARS_U + "|[\\-0-9\\u{00B7}\\u{0300}-\\u{036F}\\u{203F}-\\u{2040}]";
+  var PREFIX_START = new RegExp(PN_CHARS_BASE, "u");
+  var PREFIX_REMAINDER = new RegExp("((" + PN_CHARS + "|\\.)*(" + PN_CHARS + "))?:", "u");
```

Comment:
CodeMirror still targets browsers that don't support `u` regexps. Do you think the upper, astral character range is important to support here?

Reply:
I took the RegExp from a project that has to be exactly conformant to the spec (since improved thanks to your feedback on the character classes, thanks!), but for CodeMirror this can be weakened: I will removed the necessity to compile with "u".